include/acl_selectors.php

```diff
@@ -625,31 +631,27 @@ function _contact_link($i){ return dbesc($i['link']); }
 				intval($conv_id),
 				dbesc($search),
 				dbesc($search),
-				implode("','", $known_contacts)
+				implode("', '", $known_contacts)
 		);
-		if (dbm::is_result($r)){
+		if (dbm::is_result($r)) {
 			foreach ($r as $row) {
-				// nickname..
-				$up = parse_url($row['author-link']);
-				$nick = explode("/",$up['path']);
-				$nick = $nick[count($nick)-1];
-				$nick .= "@".$up['host'];
-				// /nickname
-				$unknow_contacts[] = array(
-					'type'    => 'c',
-					'photo'   => proxy_url($row['author-avatar'], false, PROXY_SIZE_MICRO),
-					'name'    => htmlentities($row['author-name']),
-					'id'      => '',
-					'network' => 'unknown',
-					'link'    => $row['author-link'],
-					'nick'    => htmlentities($nick),
+				$contact = get_contact_details_by_url($row['author-link'], 0);
```

Comment:
Good question. This expects all users being mentioned in the `item` table to have a public contact entry, shouldn't it?

If not, then we either have to keep the ugly hack we were previously using, or use `get_contact()` in conjunction with a function that simply retrieves contact info from a contact id, but it's supposed to be a quick auto-complete function, not probing and creating contacts.